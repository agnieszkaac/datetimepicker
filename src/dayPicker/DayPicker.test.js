import React from "react";
import { render } from "@testing-library/react";

import { Day, DayPicker } from "./";

const onPick = jest.fn();
jest.mock("moment", () => () => ({
  date: () => "1",
  isSame: (value, granularity) => true,
  daysInMonth: () => 30,
  toISOString: () => "",
}));

describe("<Day /> should", () => {
  it("match snapshot for NOT present day, NOT selected", () => {
    const { container } = render(
      <Day value={"2"} selected={false} onPick={onPick} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("match snapshot for NOT present day, selected", () => {
    const { container } = render(
      <Day value={"2"} selected={true} onPick={onPick} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("match snapshot for present day, NOT selected", () => {
    const { container } = render(
      <Day value={"1"} selected={false} onPick={onPick} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("match snapshot for present day, selected", () => {
    const { container } = render(
      <Day value={"1"} selected={true} onPick={onPick} />,
    );
    expect(container).toMatchSnapshot();
  });
});

describe("<DayPicker /> should", () => {
  it("match snapshot", () => {
    const { container } = render(
      <DayPicker value={"1"} monthLength={10} onPick={onPick} />,
    );
    expect(container).toMatchSnapshot();
  });
});
