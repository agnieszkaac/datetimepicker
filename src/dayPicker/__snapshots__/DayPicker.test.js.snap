// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day /> should match snapshot for NOT present day, NOT selected 1`] = `
<div>
  <button
    class="day is-today"
    type="button"
    value=""
  >
    1
  </button>
</div>
`;

exports[`<Day /> should match snapshot for NOT present day, selected 1`] = `
<div>
  <button
    class="day selected is-today"
    type="button"
    value=""
  >
    1
  </button>
</div>
`;

exports[`<Day /> should match snapshot for present day, NOT selected 1`] = `
<div>
  <button
    class="day is-today"
    type="button"
    value=""
  >
    1
  </button>
</div>
`;

exports[`<Day /> should match snapshot for present day, selected 1`] = `
<div>
  <button
    class="day selected is-today"
    type="button"
    value=""
  >
    1
  </button>
</div>
`;

exports[`<DayPicker /> should match snapshot 1`] = `
<div>
  <div
    class="day-picker"
  >
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
    <button
      class="day is-today"
      type="button"
      value=""
    >
      1
    </button>
  </div>
</div>
`;
